app/routes/articles/article-single/article-page.tsx

```diff
@@ -9,30 +10,87 @@ import { CardCarouselSection } from "~/components/resource-carousel";
+import { Configure, InstantSearch, useHits } from "react-instantsearch";
+import { createSearchClient } from "~/lib/create-search-client";
+import { ContentItemHit } from "~/routes/search/types";
 
 export const ArticlePage: React.FC = () => {
   const data = useLoaderData<LoaderReturnType>();
-  const tagId = data.relatedArticles?.tagId;
+
+  const { ALGOLIA_APP_ID, ALGOLIA_SEARCH_API_KEY } = data;
+
+  const searchClient = useMemo(
+    () =>
+      createSearchClient(ALGOLIA_APP_ID || "", ALGOLIA_SEARCH_API_KEY || ""),
+    [ALGOLIA_APP_ID, ALGOLIA_SEARCH_API_KEY]
+  );
+
+  const { title, articlePrimaryCategories } = data;
 
   return (
     <section className="bg-white">
       <ArticleHero {...data} />
       <ArticleContent htmlContent={data.content} />
 
-      <CardCarouselSection
-        title="Related Reading"
-        description="Explore more articles that you might find interesting."
-        resources={data.relatedArticles?.articles || []}
-        viewMoreText="More Articles"
-        viewMoreLink={`/related-articles/${tagId}`}
-        CardComponent={RelatedArticleCardWrapper}
-      />
+      {/* Related Articles Section */}
+      <InstantSearch indexName="dev_contentItems" searchClient={searchClient}>
+        <Configure
+          filters={`contentType:"Article" AND NOT title:"${title}" AND articlePrimaryCategories:"${articlePrimaryCategories[0]}"`}
+          hitsPerPage={6}
+        />
+        <CardCarouselSectionWrapper />
+      </InstantSearch>
 
       <ArticleNewsletter />
     </section>
   );
 };
 
+const CardCarouselSectionWrapper = () => {
+  const { items } = useHits<ContentItemHit>();
+
+  if (items.length === 0) {
+    return null;
+  }
+
+  return (
+    <CardCarouselSection
+      title="Related Reading"
+      description="Explore more articles that you might find interesting."
+      resources={items.map((item) => ({
+        id: String(item.rockItemId ?? item.objectID),
+        contentChannelId: "",
+        contentType: "ARTICLES",
+        name: item.title,
+        summary: item.summary,
+        image: item.coverImage.sources[0]?.uri ?? "",
+        pathname: item.url,
+        startDate: format(new Date(item.startDateTime), "d MMM yyyy"),
+        readTime: item.articleReadTime,
+        authorProps: {
+          fullName:
+            item.author.firstName && item.author.lastName
+              ? `${item.author.firstName} ${item.author.lastName}`
+              : "Christ Fellowship Team",
+          photo: {
+            uri:
+              item.author.profileImage ||
+              "http://cloudfront.christfellowship.church/GetImage.ashx?guid=A62B2B1C-FDFF-44B6-A26E-F1E213285153",
+          },
+          authorAttributes: {
+            pathname: `${item.author.firstName}-${item.author.lastName}`
+              .toLowerCase()
+              .replace(/ /g, "-"),
+          },
+        },
+      }))}
+      viewMoreText="More Articles"
+      viewMoreLink={`/articles/topics/todo`}
```

Comment:
@leoalfonso14 Looks like the "More Articles" button in the top right has yet to be linked

<img width="1633" height="813" alt="Image" src="https://github.com/user-attachments/assets/ced3cc04-b2ec-4d3b-8d07-9e80f73c5483" />

Should we just link to `/articles`?